### PR TITLE
Massachusetts rent deduction cap increase

### LINKED ISSUE #3435: Massachusetts rent deduction cap increase
Currently residents of Massachusetts are allowed a deduction of 50% of their total rental payments, capped at a maximum amount ranging from $1,500 to $3,000 , which depends on the filing status.
This cap will increase by either $1,000 or $500 depending on filing status
https://malegislature.gov/Bills/193/H4104

### PR DESCRIPTION
**Pull request by vrathi101**
Fixes #3435 

### Summary
:house_with_garden::arrow_up::moneybag:

Increased the MA rental deduction cap for income taxes by different amounts depending on filing status. This reflects the latest legislative changes as documented in `policyengine_us/parameters/gov/states/ma/tax/income/deductions/rent/cap.yaml`.

> _`rental_deduction`_
> _Up for most, half for some_
> _Fall brings tax changes_

### Walkthrough
*  Increase the maximum rental deduction for MA income taxes by different amounts for each filing status ([link](https://github.com/PolicyEngine/policyengine-us/pull/3447/files?diff=unified&w=0#diff-cbc2f9881a461e03ba791a0481e8215d98a17fd03e01dde1f3e81b3f0bcd207eL3-R11)). This change updates the MA tax parameters based on the latest legislation, using the reference https://www.mass.gov/info-details/massachusetts-tax-rates-and-tax-credits.

### CODE REVIEW

File: policyengine_us/parameters/gov/states/ma/tax/income/deductions/rent/cap.yaml [hunk 1]
```diff
@@ -1,14 +1,14 @@
 description: Maximum rental deduction for MA income taxes.
 SINGLE:
-  2011-01-01: 3_000
+  2011-01-01: 4_000
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ma/tax/income/deductions/rent/cap.yaml:3]: the increase happened in 2023, we leave the original value and add a value starting in 2023

Need new 2023 tests (the existing ones remain untouched)

File: changelog_entry.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+- bump: patch
+  changes:
+    fixed:
+    - Massachusetts rent deduction cap.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Massachusetts 2023 rent deduction cap increase.
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ma/tax/income/deductions/rent/cap.yaml [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+  2023-01-01: 4_000
 JOINT:
   2011-01-01: 3_000
+  2023-01-01: 4_000
 HEAD_OF_HOUSEHOLD:
   2011-01-01: 3_000
+  2023-01-01: 4_000
 SEPARATE:
   2011-01-01: 1_500
+  2023-01-01: 2_000
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ma/tax/income/deductions/rent/cap.yaml:16]: need to add the legal reference and a test for 2023

File: policyengine_us/parameters/gov/states/ma/tax/income/deductions/rent/cap.yaml [hunk 4]
```diff
@@ -1,14 +1,19 @@
 description: Maximum rental deduction for MA income taxes.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ma/tax/income/deductions/rent/cap.yaml:1]:
```suggestion
description: Massachusetts capes the rental deduction at this amount, based on the filing status. 
```

File: policyengine_us/parameters/gov/states/ma/tax/income/deductions/rent/cap.yaml (hunk 3 above)
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ma/tax/income/deductions/rent/cap.yaml:16]: change label to:
`label: Massachusetts rental deduction cap`

File: policyengine_us/parameters/gov/states/ma/tax/income/deductions/rent/cap.yaml [hunk 5]
```diff
@@ -1,19 +1,24 @@
...
 HEAD_OF_HOUSEHOLD:
   2011-01-01: 3_000
+  2023-01-01: 4_000
 SEPARATE:
   2011-01-01: 1_500
+  2023-01-01: 2_000
 metadata:
   reference:
     # Legislation only mentions the $3,000.
     - title: "Taxable income: adjusted gross income less deductions and exemptions (a)(9)"
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ma/tax/income/deductions/rent/cap.yaml:20]:
```suggestion
    - title: Massachusetts General Laws (I)(IX)(62)(3)(B)(a)(9)"
```

File: policyengine_us/parameters/gov/states/ma/tax/income/deductions/rent/cap.yaml [hunk 6]
```diff
@@ -1,19 +1,24 @@
...
+  2023-01-01: 4_000
 HEAD_OF_HOUSEHOLD:
   2011-01-01: 3_000
+  2023-01-01: 4_000
 SEPARATE:
   2011-01-01: 1_500
+  2023-01-01: 2_000
 metadata:
   reference:
     # Legislation only mentions the $3,000.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ma/tax/income/deductions/rent/cap.yaml:19]:
```suggestion
    # Legal code only mentions the $3,000.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**